src/shogun/mathematics/linalg/ratapprox/tracesampler/TraceSampler.h

```diff
@@ -17,12 +18,12 @@ template<class T> class SGVector;
 /** @brief Abstract template base class that provides an interface for sampling
  * the trace of a linear operator using an abstract sample method
  */
-class CTraceSampler : public CSGObject
+class CTraceSampler : public CSGObject, protected CRandom
 {
 public:
 	/** default constructor */
 	CTraceSampler()
-	: CSGObject()
+	: CSGObject(), CRandom(this)
```

Comment:
Could we somehow create something that doesnt rely on this ctor (like calling default ctor)? Likely to be forgotten :D
I am not sure it is possible though

Reply:
using a mixin that has an CRandom object (as you suggested) will solve this.